play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html

```diff
@@ -31,36 +33,28 @@ <h2 class="govuk-cookie-banner__heading govuk-heading-m">
           <div class="govuk-cookie-banner__content">@message.content match {
               case NonEmptyHtml(html) => {@html}
               case NonEmptyText(text) => {<p class="govuk-body">@text</p>}
-              case _ => {}}</div>
+              case _ => { }}</div>
         </div>
       </div>
       @message.actions.map { actions =>
         <div class="govuk-button-group">
           @for(action <- actions) {
-            @if(action.href.exists(_.nonEmpty)) {
-               @if(action.inputType.exists(_ == "button")) {
-                   @govukButton(Button(
-                       content = Text(action.text),
-                       inputType = Some("button"),
-                       classes = action.classes,
-                       attributes = action.attributes,
-                       href = action.href
-                   ))
-               } else {
-                   <a class="@toClasses("govuk-link", action.classes)" href="@{action.href}"@toAttributes(action.attributes)>@action.text</a>
-               }
+            @if(action.href.getOrElse("").isEmpty || action.inputType.exists(_ == "button"))  {
```

Comment:
small thing to see if I missed something but why did we change this to action.href.getOrElse("").isEmpty rather than action.href.exists(_.nonEmpty) seems like more steps where exists would short circuit straight away on None

Reply:
I think we're trying to catch where `href` is `None` or `Some("")`, which using `exists` would probably be `!action.href.exists(_.nonEmpty)` which is arguably more mind-bending 😅 
Broadly speaking though, there are many places where we're having to code defensively purely to workaround the ambiguous JS model (see `Option[Seq[...]]`) and/or our own generative tests that populate those models with illogical inputs - it could be worth looking at some more strict typing here, or `require`ing some preconditions to make these cases go away... or it could just turn out to be yak-shaving 🤷

Reply:
ah, I get you, I totally misread the changed logic of it 🤦‍♂️